src/connectService.ts

```diff
@@ -138,20 +165,29 @@ class ConnectService extends IframeService {
     }
 
     this.setActiveAccount(accountId)
+
+    if (migration && legacyDid) {
+      assert.isDefined(this.idx, 'IDX instance must be defined')
+      //TODO handle partial migration failures
+      const profile = get3BoxProfile(legacyDid)
+      //TODO which definition are we using? 
+      const existing = (await this.idx.get('basicProfile')) || {}
+      await this.idx.set('basicProfile', Object.assign(existing, profile))
```

Comment:
https://github.com/ceramicstudio/js-idx-tools/blob/master/src/schemas/BasicProfile.json

Reply:
Here are the TypeScript types for this: https://github.com/ceramicstudio/js-idx-constants/blob/master/src/index.ts#L3-L29